#!/usr/bin/env python
PACKAGE = 'docking'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("world_frame",    str_t,    0, "Name of world frame",  "world")
gen.add("robot_frame",    str_t,    0, "Name of robot frame",  "base_link")
gen.add("dock_frame",    str_t,    0, "Name of dock frame",  "dock_truth")
gen.add("cmd_vel_topic",    str_t,    0, "Name of cmd_vel topic",  "cmd_vel")
gen.add("perform_detection",   bool_t,   0, "Activate the detection node",  True)
gen.add("perform_planning",   bool_t,   0, "Activate the planning node",  True)
gen.add("dock_pose_topic",    str_t,    0, "Topic of calculated dock pose",  "line_detection_node/docking/dock_pose")
gen.add("plan_topic",    str_t,    0, "Topic of calculated plan",  "docking/plan")
gen.add("publish_twist",   bool_t,   0, "Publish the twist commands to execute",  False)
gen.add("goal_dist_tolerance", double_t, 0, "Max Goal Distance Error Tolerance (m)", 0.03, 0.01, 1)
gen.add("goal_orientation_tolerance", double_t, 0, "Max Goal Orientation Error Tolerance (rad)", 0.1, 0.01, 1)
gen.add("time_step", double_t, 0, "Time per step in pose path", 0.1, 0.1, 1)


gen.add("frequency", int_t, 0, "Controller frequency", 10, 1, 30)

exit(gen.generate(PACKAGE, "docking", "ControllerNode"))
